Core/include/Acts/TrackFinding/CombinatorialKalmanFilter.hpp

```diff
@@ -234,23 +245,27 @@ class CombinatorialKalmanFilter {
   ///
   /// The CombinatorialKalmanFilter Actor does not rely on the measurements to
   /// be sorted along the track.
-  template <typename source_link_t, typename parameters_t,
+  template <typename source_link_accessor_t, typename parameters_t,
             typename calibrator_t, typename measurement_selector_t>
   class Actor {
    public:
     using TipState = CombinatorialKalmanFilterTipState;
     using BoundState = std::tuple<BoundTrackParameters, BoundMatrix, double>;
     using CurvilinearState =
         std::tuple<CurvilinearTrackParameters, BoundMatrix, double>;
+    // The source link container type
+    using SourceLinkContainer = typename source_link_accessor_t::container_type;
+    // The SourceLink type fulfilling the @c SourceLinkConcept
+    using SourceLink = typename source_link_accessor_t::value_type;
     /// Broadcast the result_type
-    using result_type = CombinatorialKalmanFilterResult<source_link_t>;
+    using result_type = CombinatorialKalmanFilterResult<SourceLink>;
 
     /// The target surface
     const Surface* targetSurface = nullptr;
 
-    /// Allows retrieving measurements for a surface
-    const std::unordered_map<GeometryIdentifier, std::vector<source_link_t>>*
-        inputMeasurements;
+    /// The input measurements which should have the geometry identifier as the
+    /// key type and are accessed with the help of the accessor
+    const SourceLinkContainer* inputMeasurements;
```

Comment:
Do we need this anymore? I think this should be encapsulated in the accessor itself.

Reply:
Right, it should be more straghtforward to include the container in the accessor and just rename the 'accessor` to `container`.

Reply:
I included a pointer to the source links in the accessor instead of the source links themselves to avoid costly copy (so it's still an `accessor`). 

The input to the `findTracks` is still a reference to the `sourcelinks` and a pointer to the `sourcelinks` is passed to the accessor inside the `findTracks`.  Alternatively, we could use the accessor directly as the input (assuming the pointer to the  source links is already set properly by the user), but I'm afraid this is less safe since we lose track of the lifetime of the source links the pointer is pointing to.

What do you think?

Reply:
This is a very valid point. Let's do it like this, and possibly revisit when we have a clearer picture on the concrete integration.